challenge_2/go/erocs/find_singleton.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+)
+
+func FindSingleton(int_array []int) (found int) {
+	singles := make(map[int]int)
+	repeats := make(map[int]int)
```

Comment:
This is one less loop iteration at the cost of one allocation. Different trade-offs. Realistically, these maps are being used as Sets and _repeats_ doesn't need to bother counting.